### PR TITLE
remove constraint on Thermodynamics

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

https://github.com/CliMA/Thermodynamics.jl/commit/1701c9aa474f43814e5d16373606f5ffd0eb6763 fixes the definition of virtual DSE for surface fluxes, and bumps a patch version, so the constraint on Thermodynamics is not needed anymore. Also updates tests and bump patch version.

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by Sbozzolo** [comment]
I think it's better be on the safe side and put 0.11.5

File: Project.toml [hunk 1]
```diff
@@ -19,5 +19,5 @@ CreateParametersExt = "CLIMAParameters"
 CLIMAParameters = "0.8"
 DocStringExtensions = "0.8, 0.9"
 RootSolvers = "0.2, 0.3, 0.4"
-Thermodynamics = "< 0.11.4"
+Thermodynamics = "0.11"
```
Comment (Sbozzolo) [inline Project.toml:22]:
```suggestion
Thermodynamics = "0.11.5"
```

**Comment by Sbozzolo** [comment]
CI is failing :'(

**Comment by szy21** [comment]
Hmm the recent change in Thermodynamics should cause behavior changes, but I thought it wouldn't break the tests. It might be just that we need to update the tests. Let me check.

**Comment by szy21** [comment]
@akshaysridhar Given we change the definition of virtual DSE (from cp_m to cp_d), the ΔDSEᵥ in test_cases is not accurate anymore. All the failing cases in `test_cases` are near neutral conditions, so I guess it is because ΔDSEᵥ changes sign from the definition change. I can just modify the tests or remove them. What do you think?

**Comment by akshaysridhar** [comment]
I think I agree that these test cases are linked to specific commits within the ClimaAtmos package (the classification was based on the tolerances and older versions of ClimaAtmos+SF - these are prior to Float32 improvements in SF and other physics changes within Atmos) - it makes little sense to maintain checks for those specific thermodynamic inputs; however I think we should track a separate issue then to catch edge cases that rely on our tolerance definitions.

**Comment by szy21** [comment]
> I think I agree that these test cases are linked to specific commits within the ClimaAtmos package (the classification was based on the tolerances and older versions of ClimaAtmos+SF - these are prior to Float32 improvements in SF and other physics changes within Atmos) - it makes little sense to maintain checks for those specific thermodynamic inputs; however I think we should track a separate issue then to catch edge cases that rely on our tolerance definitions.

These cases are still edge cases, they just change from slightly unstable to slightly stable. So maybe we can keep them but change the sign for the test?

**Comment by szy21** [comment]
@akshaysridhar I update the tests with the new numbers. Would you like to take a look again? Some of them are not edge cases anymore, so we can also remove them if you want.

File: Project.toml [hunk 2]
```diff
@@ -19,5 +19,5 @@ CreateParametersExt = "CLIMAParameters"
 CLIMAParameters = "0.8"
 DocStringExtensions = "0.8, 0.9"
 RootSolvers = "0.2, 0.3, 0.4"
-Thermodynamics = "< 0.11.4"
+Thermodynamics = "0.11.5"
```
Comment (charleskawczynski) [inline Project.toml:22]: My general preference is to exclude patch numbers here. That's afterall how Julia's compat helper makes recommendations.

**Review by charleskawczynski — APPROVED** [review]

File: Project.toml (hunk 2 above)
Comment (szy21) [reply Project.toml:22]: @Sbozzolo any suggestion?

Comment (Sbozzolo) [reply Project.toml:22]: Because of the changes, the new version will not be compatible with Thrmodynamics `0.11.0-0.11.4`, so it should be `0.11.5`.

Comment (szy21) [reply Project.toml:22]: Ah, that's right!
